wazuh/wazuh-kibana-app
# [Redesign add agent] Add register agent command generator

### Description
This PR resolves issue #5455 

### How to test

- Check implementation
- Check documentation
  -  [Command generator documentation](https://github.com/wazuh/wazuh-kibana-app/blob/8c3bcf9fcf8746255a634a9cb5ca96440fe31d8b/public/controllers/register-agent/core/register-commands/README.md)
  - [Hook documentation](https://github.com/wazuh/wazuh-kibana-app/blob/8c3bcf9fcf8746255a634a9cb5ca96440fe31d8b/public/controllers/register-agent/core/register-commands/README.md)
- This PR doesn't have manual testing 

**Important:** The OS definitions added in the `config` folder are only an example to show the configuration that receives the command generator
The OS definitions and the optional parameters will be adjusted in the integration issue #5547
